File: backend/cs_api/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace cs_api.Models
{
  [Table("product")]
  public class Product
  {
    public int Id { get; set; }

    [Required]
    public string Name { get; set; } = "";

    [Required]
    public decimal Price { get; set; }

    [Required]
    public DateTime Expiration { get; set; }

    [Required]
    public DateTime CreatedAt { get; set; }

    // Brazilian Reais  
    public decimal BRL { get; set; }

    // Euro
    public decimal EUR { get; set; }

    // Canadian dollar
    public decimal CAD { get; set; }

    // Mexican Pesos
    public decimal MXN { get; set; }

    // Libras
    public decimal GBP { get; set; }
  }
}

File: mobile/lib/domain/models/product.dart
class Product {
  //
  Product({
    required this.id,
    required this.name,
    required this.price,
    required this.expiration,
    required this.createdAt,
    this.brl,
    this.eur,
    this.cad,
    this.mxn,
    this.gbp,
  });

  late int id;
  late String name;
  late double price;
  late DateTime expiration;
  late DateTime createdAt;

  // Brazilian Reais
  double? brl;
  // Euro
  double? eur;
  // Canadian dollar
  double? cad;
  // Mexican Pesos
  double? mxn;
  // Libras
  double? gbp;
}
